pkg/antctl/raw/multicluster/create/clusterclaim.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package create
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/pkg/antctl/raw"
+	multiclusterscheme "antrea.io/antrea/pkg/antctl/raw/multicluster/scheme"
+)
+
+type clusterClaimOptions struct {
+	namespace  string
+	clusterSet bool
+}
+
+var clusterClaimOpt *clusterClaimOptions
+
+var clusterClaimExamples = strings.Trim(`
+# Create a ClusterClaim for a leader or member cluster
+  $ antctl mc create clusterclaim <CLUSTER_ID> -n <NAMESPACE>
+# Create a ClusterClaim for a ClusterSet
+  $ anctcl mc create clusterclaim <CLUSTER_ID> -n <NAMESPACE> --clusterset
+`, "\n")
+
+func (o *clusterClaimOptions) validateAndComplete() error {
+	if o.namespace == "" {
+		return fmt.Errorf("the Namespace cannot be empty")
+	}
+
+	return nil
+}
+
+func NewClusterClaimCmd() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "clusterclaim",
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Create a ClusterClaim in a leader or member cluster",
+		Long:    "Create a ClusterClaim in a leader or member cluster",
+		Example: clusterClaimExamples,
+		RunE:    clusterClaimRunE,
+	}
+
+	o := &clusterClaimOptions{}
+	clusterClaimOpt = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of the ClusterClaim")
+	command.Flags().BoolVarP(&o.clusterSet, "clusterset", "", false, "If present, a ClusterClaim of ClusterSet will be created")
+
+	return command
+}
+
+func clusterClaimRunE(cmd *cobra.Command, args []string) error {
+	if err := clusterClaimOpt.validateAndComplete(); err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("exactly one NAME is required, got %d", len(args))
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	clusterClaimValue := args[0]
```

Comment:
Do you think it is strange to use arg[0] typically for name as the clusterClaimValue?

I think we should make it clear. Paste my earlier comments:

> We just need to add two parameters one like "--clusterset <CLUSTERSET_ID>", one like "--cluster-id <CLUSTER_ID>". The first Claim uses CLUSTER_ID as name, and the second uses CLUSTERSET_ID as name.

Reply:
Yes. I feel strange, too. So I removed arg[0] and set ` cobra.MaximumNArgs(0)`, both the ClusterClaim for ClusterSet and ClusterClaim for the leader or member cluster will get ID from flags.